sympy/solvers/tests/test_ode.py

```diff
@@ -3037,12 +3061,77 @@ def test_2nd_power_series_regular():
         + O(x**6))
 
     eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 - S(1)/4)*f(x)
-    assert dsolve(eq) == Eq(f(x), C1*(x**4/24 - x**2/2 + 1)/sqrt(x) +
+    assert dsolve(eq, hint='2nd_power_series_regular') == Eq(f(x), C1*(x**4/24 - x**2/2 + 1)/sqrt(x) +
         C2*sqrt(x)*(x**4/120 - x**2/6 + 1) + O(x**6))
 
     eq = x*(f(x).diff(x, 2)) - f(x).diff(x) + 4*x**3*f(x)
     assert dsolve(eq) == Eq(f(x), C2*(-x**4/2 + 1) + C1*x**2 + O(x**6))
 
+def test_2nd_linear_bessel_equation():
+    eq = x**2*(f(x).diff(x, 2)) + x*(f(x).diff(x)) + (x**2 - 4)*f(x)
+    sol = Eq(f(x), C1*besselj(2, x) + C2*bessely(2, x))
+    sols = constant_renumber(sol)
+    assert classify_ode(eq) == ('2nd_linear_bessel', '2nd_power_series_regular')
```

Comment:
We don't need to test `classify_ode` here since using `hint='2nd_linear_bessel'` below would raise if it doesn't match `2nd_linear_bessel`.

Reply:
okay

Reply:
This line isn't needed

Reply:
All the `classifyode` lines aren't needed if using `dsolve(hint=)`